controllers/servers/host_definer/storage_manager/host_definer_server.py

```diff
@@ -24,26 +23,21 @@ def define_host(self, request):  # pylint: disable=invalid-name
             with get_agent(array_connection_info, array_type).get_mediator() as array_mediator:
 
                 try:
-                    found_host_name, _ = array_mediator.get_host_by_host_identifiers(initiators)
-                    logger.debug("host found : {}".format(found_host_name))
+                    self._get_host_name(initiators, array_mediator)
                 except HostNotFoundError:
                     logger.debug("host was not found. creating a new host with initiators: {0}".format(initiators))
                     try:
-                        array_mediator.create_host(host_name, initiators, connectivity_type)
+                        array_mediator.create_host(host_name, initiators, request.connectivity_type)
```

Comment:
I know it's not entirely related to your PR, but can you move [this](https://github.com/IBM/ibm-block-csi-driver/blob/2f2c3265f98c77f19079c87e027abb1d44044a3f/controllers/servers/host_definer/storage_manager/host_definer_server.py#L29-L33) code into `_create_new_host` function, for clarity?

Reply:
since that code block contains `return` but only for the exception, it's not a simple task. I prefer not to do it.

Reply:
ok whatever is comfortable for you I am good with it